WorkAndFun/src/main/java/pl/infoshare/announcements/AnnouncementSearchService.java

```diff
@@ -0,0 +1,110 @@
+package pl.infoshare.announcements;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class AnnouncementSearchService extends AnnouncementService {
+
+    public void displaySearchMenu() throws ReturnToMenuException {
+        System.out.println("1. Wyszukaj oferty na wykonanie usług.");
+        System.out.println("2. Wyszukaj oferty na zapotrzebowanie usług.");
+        System.out.println("0. Powrót do menu");
+        String userInput = scanner.nextLine();
+        switch (userInput) {
+            case "1":
+                System.out.println("Wybrałeś wyszukiwanie wykonywania usług.");
+                searchAnnoucement(OfferType.SERVICE_OFFER);
+                break;
+            case "2":
+                System.out.println("Wybrałeś wyszukiwanie zapotrzebowania usług.");
+                searchAnnoucement(OfferType.SERVICE_DEMAND);
+                break;
+            case "0":
+                break;
+            default: {
+                System.out.println("Podałeś zły parametr!");
+                displaySearchMenu();
+                break;
+            }
+        }
+    }
+
+    public void searchAnnoucement(OfferType offerType) throws ReturnToMenuException {
+        var announcementList = announcementRepository.findAll();
+        String userInput = getInputFromUser("Wpisz co chcesz wyszukać " +
+                        "1 - Lokalizacja 2 - Kategoria 3 - Opis :"
+                , "[1-3]{1}", "Nie ma pola o takim numerze, wprowadz poprawny numer pola do edycji");
+
+        if (userInput == null) {
+            throw new ReturnToMenuException("Wybrałes powrót do menu");
+        }
+        switch (Integer.parseInt(userInput)) {
+            case 1:
+                System.out.println("Podaj lokalizację ogłoszenia");
+                var announcementsByLocalisation = searchByLocalisation(announcementList, offerType);
```

Comment:
```suggestion
                var announcementsByLocation = searchByLocalisation(announcementList, offerType);
```
chyba chodzi o Location

Reply:
zmienione